api/envoy/service/status/v3/csds.proto

```diff
@@ -42,15 +42,20 @@ enum ConfigStatus {
   // Management server has sent the config to client and received ACK.
   SYNCED = 1;
 
-  // Config is not sent.
+  // Config is not sent from the management server; or the client haven't
+  // received the requested config.
   NOT_SENT = 2;
 
   // Management server has sent the config to client but hasn’t received
   // ACK/NACK.
   STALE = 3;
 
-  // Management server has sent the config to client but received NACK.
+  // Management server has sent the config to client but received NACK; or the
+  // client has received the config but replied with NACK.
   ERROR = 4;
+
+  // The client has received the config and repied with ACK.
```

Comment:
I'm almost thinking we want separate status enum messages or very explicit field naming to help make clear which one (client vs. server status) we're talking about.

Reply:
Yeah, seeing these comments, I guess I do think that having a separate set of enums would be clearer.  We can call them something like `CLIENT_REQUESTED`, `CLIENT_ACKED`, and `CLIENT_NACKED`.

We can either add these as a new values in this existing enum, or we can add a whole separate enum field for use in clients.

Reply:
Adding a new proto enum struct named ClientSideStatusConfig. It's a bit verbose, but I think it needs to have a very clear naming.